### ISSUE TITLE
Name collision if there is another blueprint with endpoint name "admin"

### ISSUE THREAD

**Issue by rj79** [flask-admin/flask-admin#1815]
Hi!

If there is already a blueprint with endpoint name "admin" there will be a name collision even if you pass something else as endpoint parameter to the Admin constructor.

Here is a minimal application to show the problem:

```python
# Minimal example to show name clashes between flask-admin and blueprints
# with name "admin".
from flask import Blueprint, Flask
from flask_admin import Admin

app = Flask(__name__)

# This blueprint claims the endpoint name 'admin'
bp = Blueprint('admin', __name__)

# Accessed with /admin/
@bp.route('/')
def show():
    return "Blueprint"

app.register_blueprint(bp, url_prefix='/admin')

@app.route('/')
def index():
    # Just to show which routes are registered
    return ' '.join(sorted([str(rule) for rule in app.url_map.iter_rules()]))

# Error if trying to access /b2 or /b3 because the corresponding templates/bootstrap[2|3]/static.html 
# calls url_for('admin.static'), instead of honoring the endpoint parameter. 
# This clashes with the admin blueprint above.
flask_admin2 = Admin(app, url='/b2', endpoint='bootstrap2', template_mode='bootstrap2')
flask_admin3 = Admin(app, url='/b3', endpoint='bootstrap3', template_mode='bootstrap3')

app.run()
```

**Comment by luyangong** [comment]
I test it. But everything is OK!

**Comment by rj79** [comment]
Start the server and then try to access http://localhost:5000/b2 or http://localhost:5000/b3. 

Expected behavior:
The flask admin interface is shown.

**Actual behavior:**
Page shows:
Internal Server Error
The server encountered an internal error and was unable to complete your request. Either the server is overloaded or there is an error in the application.

**Console output**:
 * Serving Flask app "example" (lazy loading)
 * Environment: production
   WARNING: Do not use the development server in a production environment.
   Use a production WSGI server instead.
 * Debug mode: off
 * Running on http://127.0.0.1:5000/ (Press CTRL+C to quit)
[2019-03-24 21:56:30,298] ERROR in app: Exception on /b2/ [GET]
Traceback (most recent call last):
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/app.py", line 2292, in wsgi_app
    response = self.full_dispatch_request()
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/app.py", line 1815, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/app.py", line 1718, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/app.py", line 1813, in full_dispatch_request
    rv = self.dispatch_request()
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask_admin/base.py", line 69, in inner
    return self._run_view(f, *args, **kwargs)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask_admin/base.py", line 368, in _run_view
    return fn(self, *args, **kwargs)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask_admin/base.py", line 452, in index
    return self.render(self._template)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask_admin/base.py", line 308, in render
    return render_template(template, **kwargs)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/templating.py", line 135, in render_template
    context, ctx.app)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/templating.py", line 117, in _render
    rv = template.render(context)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/jinja2/asyncsupport.py", line 76, in render
    return original_render(self, *args, **kwargs)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/jinja2/environment.py", line 1008, in render
    return self.environment.handle_exception(exc_info, True)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask_admin/templates/bootstrap2/admin/index.html", line 1, in top-level template code
    {% extends 'admin/master.html' %}
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask_admin/templates/bootstrap2/admin/master.html", line 1, in top-level template code
    {% extends admin_base_template %}
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask_admin/templates/bootstrap2/admin/base.html", line 14, in top-level template code
    {% block head_css %}
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask_admin/templates/bootstrap2/admin/base.html", line 15, in block "head_css"
    <link href="{{ admin_static.url(filename='bootstrap/bootstrap2/swatch/{swatch}/bootstrap.min.css'.format(swatch=config.get('FLASK_ADMIN_SWATCH', 'default')), v='2.3.2') }}" rel="stylesheet">
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/jinja2/runtime.py", line 579, in _invoke
    rv = self._func(*arguments)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask_admin/templates/bootstrap2/admin/static.html", line 2, in template
    {{ get_url('admin.static', *varargs, **kwargs) }}
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask_admin/base.py", line 390, in get_url
    return url_for(endpoint, **kwargs)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/helpers.py", line 356, in url_for
    return appctx.app.handle_url_build_error(error, endpoint, values)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/app.py", line 2061, in handle_url_build_error
    reraise(exc_type, exc_value, tb)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/_compat.py", line 35, in reraise
    raise value
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/flask/helpers.py", line 345, in url_for
    force_external=external)
  File "/home/robjo/code/example/.venv/lib/python3.6/site-packages/werkzeug/routing.py", line 2182, in build
    raise BuildError(endpoint, values, method, self)
werkzeug.routing.BuildError: Could not build url for endpoint 'admin.static' with values ['filename', 'v']. Did you mean 'static' instead?
127.0.0.1 - - [24/Mar/2019 21:56:30] "GET /b2/ HTTP/1.1" 500 -
`

**Comment by luyangong** [comment]
Sorry for my misunderstanding. Maybe it is a feature. 
![image](https://user-images.githubusercontent.com/31658620/55086313-23353a80-50e3-11e9-874c-a444c7bb2a1b.png)

**Comment by nk9** [comment]
I spent _days_ trying to debug this myself before realizing that I had to change the name of my 'admin' blueprint. The errors were very confusing, it wasn't clear what was going on. I was shocked when I figured out what the problem really was.

It seems crazy that Flask-Admin itself would have a blueprint with this name. Surely the first name most developers would choose for the blueprint which contains the admin panel is 'admin'!

I think Flask-Admin should change the name of its admin blueprint to save headaches for all the library's users. I propose 'flask_admin'.
